### DISCUSSION TITLE
Custom Validator pull inventory objects

### DISCUSSION THREAD

**Discussion by settlej** [netbox-community/netbox#11681] [Q&A]
How do you get a list of all dcim/locations in a custom validator?  I created a custom field and want to make sure every time a new location is created it has a unique id associated to it.  I tried:

from extras.validators import CustomValidator
from dcim.models import Location

```
class UniqueLocationID(CustomValidator):
    def validate(self, location):
         print(Location.objects.all())
```

I get an error about the apps are not fully loaded.

**Comment by kkthxbye-code** [comment]
You need to do model imports inside the validate method because, as it tells you, apps are not loaded yet when the settings file is executed.
